src/code/z_sub_s.c

```diff
@@ -525,9 +565,92 @@ void SubS_DrawShadowTex(Actor* actor, GameState* gameState, u8* tex) {
     CLOSE_DISPS(gfxCtx);
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sub_s/func_8013D0E0.s")
+/**
+ * Computes the rotation based on the options and target rotation value
+ *
+ * @param[in,out] rot the computed rotation
+ * @param[in] rotMax the max rotation in binary angles
+ * @param[in] target the target rotation value
+ * @param[in] slowness how slow to rotate, the larger the number the slower the rotation
+ * @param[in] rotAdjMin the minimun rotation adjustment in degrees
+ * @param[in] rotAdjMax the maximum rotation adjustment in degrees
+ */
+s16 SubS_ComputeTurnToPointRot(s16* rot, s16 rotMax, s16 target, f32 slowness, f32 rotAdjMin, f32 rotAdjMax) {
+    s16 prevRot = *rot;
+    f32 rotAdj;
+    f32 prevRotAdj;
+
+    rotAdj = (f32)(target - *rot) * (360.0f / 65536.0f);
+    rotAdj *= gFramerateDivisorHalf;
+    prevRotAdj = rotAdj;
+    if (rotAdj >= 0.0f) {
+        rotAdj /= slowness;
+        rotAdj = CLAMP(rotAdj, rotAdjMin, rotAdjMax);
+        *rot += (s16)((rotAdj * 65536.0f) / 360.0f);
+        if (prevRotAdj < rotAdjMin) {
+            *rot = target;
+        }
+        if (rotMax != 0) {
+            *rot = CLAMP(*rot, -rotMax, rotMax);
+        }
+    } else {
+        rotAdj = (rotAdj / slowness) * -1.0f;
+        rotAdj = CLAMP(rotAdj, rotAdjMin, rotAdjMax);
+        *rot -= (s16)((rotAdj * 65536.0f) / 360.0f);
+        if (-rotAdjMin < prevRotAdj) {
+            *rot = target;
+        }
+        if (rotMax != 0) {
+            *rot = CLAMP(*rot, -rotMax, rotMax);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_sub_s/func_8013D2E0.s")
+    return prevRot - *rot;
+}
+
+/**
+ * Computes the necessary HeadRot and TorsoRot adjustments to smoothly turn an actors's head and torso to a point
+ *
+ * @param[in] point the point to turn to
+ * @param[in] focusPos the actor's focus postion
+ * @param[in] shapeRot the actor's shape rotation
+ * @param[in,out] turnTarget the itermediate target step that headRot and torsoRot step towards
+ * @param[in,out] headRot the computed head rotation
+ * @param[in,out] torsoRot the computed torso rotation
+ * @param[in] options various options to adjust how the actor turns, see `SubS_ComputeTurnToPointRot`
+ *
+ * options are a u16[4][4] where
+ *  the rows are headRotX, headRotY, torsoRotX, torsoRotY and
+ *  the cols are rotMax, slowness, rotAdjMin, rotAdjMax
```

Comment:
Oh for goodness' sake... what a horrible thing to program. I'm tempted to suggest enums, even though it's literally for one function, just to make this clearer.